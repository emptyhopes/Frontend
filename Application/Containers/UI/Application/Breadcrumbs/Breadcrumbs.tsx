import styled from "styled-components";

// import { useLocation, useNavigate } from "react-router-dom";

// const StyledContent = styled.div``;
const StyledLi = styled.li``;
const StyledButton = styled.button``;

const Breadcrumbs: React.FunctionComponent = () => {
  // const location = useLocation();
  // const history = useNavigate();

  return (
    <>
      <StyledLi>
        <StyledButton></StyledButton>
      </StyledLi>
    </>
  );
};

export { Breadcrumbs };
