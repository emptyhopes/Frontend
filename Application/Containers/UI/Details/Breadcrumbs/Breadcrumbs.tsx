import styled from "styled-components";

import { Link, useLocation } from "react-router-dom";

const StyledContent = styled.div`
  display: flex;
  margin: 20px 0;

  & li + li {
    margin: 0 0 0 10px;
  }
`;

const StyledLi = styled.li`
  display: flex;

  & a {
    color: blue;
  }

  & div {
    margin: 0 0 0 10px;
  }
`;

const Breadcrumbs: React.FunctionComponent = (): React.ReactElement => {
  const { pathname } = useLocation();
  const pathnames = pathname.split("/").filter((value) => value);

  const Uppercase = (value: string) => {
    value = value.toUpperCase().slice(0, 1) + value.slice(1, value.length);

    return value;
  };

  return (
    <>
      <StyledContent>
        {pathnames.length > 0 ? (
          <StyledLi>
            <Link to={"/"}>Home</Link>
            <div>/</div>
          </StyledLi>
        ) : (
          <></>
        )}

        {pathnames.map((name, index) => {
          const route = pathnames.slice(0, index + 1).join("/");
          const last = index === pathnames.length - 1;

          return last ? (
            <StyledLi key={name}>{Uppercase(name)}</StyledLi>
          ) : (
            <StyledLi key={name}>
              <Link to={route}>{Uppercase(name)}</Link>
              <div>/</div>
            </StyledLi>
          );
        })}
      </StyledContent>
    </>
  );
};

export { Breadcrumbs };
