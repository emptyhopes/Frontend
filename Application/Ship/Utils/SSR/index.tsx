import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
