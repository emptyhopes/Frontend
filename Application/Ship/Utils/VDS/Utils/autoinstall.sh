#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
