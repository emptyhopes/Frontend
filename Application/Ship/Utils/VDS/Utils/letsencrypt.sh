#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname