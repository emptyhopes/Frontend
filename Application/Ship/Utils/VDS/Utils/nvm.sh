#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
