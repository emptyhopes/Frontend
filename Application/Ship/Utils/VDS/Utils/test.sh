#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru