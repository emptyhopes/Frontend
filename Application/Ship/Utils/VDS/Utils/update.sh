#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes