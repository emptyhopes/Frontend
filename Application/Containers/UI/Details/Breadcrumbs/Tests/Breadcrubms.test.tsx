import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Breadcrumbs } from "@/Application/Containers/UI/Details/Breadcrumbs/Breadcrumbs";

test("Breadcrumbs", () => {
  const { container } = RenderTests(<Breadcrumbs />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
