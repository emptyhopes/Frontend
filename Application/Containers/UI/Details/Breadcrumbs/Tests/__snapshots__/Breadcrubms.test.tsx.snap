// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 20px 0;
}

.c0 li + li {
  margin: 0 0 0 10px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
